app/components/application_component.rb

```diff
@@ -16,13 +16,53 @@ class ApplicationComponent < ViewComponent::Base
   #   end
   # end
 
-  def initialize(*_args, id: nil, classes: '', **_kwargs)
+  def initialize(*_args, id: nil, classes: '', theme: nil, **_kwargs)
     @id = id
     @classes = class_names(classes)
     add_classes(self.class.name.underscore.dasherize.parameterize)
+    add_theme(theme)
   end
 
   def add_classes(classes)
     @classes = class_names(@classes, classes)
   end
+
+  # may put this in a card component base class but want to avoid too many levels of inheritance if possible
+  # tempted to cascade the theme down
+  def add_theme(theme)
```

Comment:
as not all (most?) of the components won't be themable in quite this way, would this be better in a mixin if you don't want to add a base class for cards?